### DISCUSSION TITLE
API rule accessing the same field twice. What am i doing wrong?

### DISCUSSION THREAD

**Discussion by nerdlibfront** [pocketbase/pocketbase#5701] [Q&A]
Pocketbase: 0.22.21
JS Client: 0.21.5

I really like Pocketbase so far, but i am getting to more complex API rules the more i use it.

The usecase i have is more complex, it's about group memberships and leader roles in their etc., but i simplified it for the purpose of this question.
So lets assume the following dataset:

<img width="1183" alt="Screenshot 2024-10-18 at 21 28 09" src="https://github.com/user-attachments/assets/4b697c27-b7ee-4dd8-a7e7-8ece788889f5">

`user` and `leaders` are a single and a multiple reference to the default `users` table.

A logged in user should only see the records that:
a) have the logged in user in the `leaders` list
b) have the user from `user` field also in the `leaders` list

For that i thought the following rule should work:
`@request.auth.id != "" && leaders.id ?= @request.auth.id && leaders.id ?= user.id`

But this results in zero results.
So is it just not possible to reference the leaders field twice here?
Or am i just doing sth. wrong here?

Thanks for helping out!

**Comment by ganigeorgiev** [comment]
This is similar to https://github.com/pocketbase/pocketbase/discussions/5290, https://github.com/pocketbase/pocketbase/discussions/4319 and probably others.

It doesn't work because the rule evaluate in a single run and both `leaders.id` references a single row value, aka. it can't be 2 different things at the same time (_see the pseudo SQL examples in the linked discussions_).

I don't exactly understand your collections structure and requirements, but one possible workaround could be to use [back-relation](https://pocketbase.io/docs/working-with-relations/#back-relations):
```js
@request.auth.id != "" &&
// self join the groups based on the current user existing in the leaders field
@request.auth.groups_via_leaders.id ?= id &&
leaders.id ?= user
```

Another possible workaround could be to apply a "LIKE" search in the raw serialized json field value using the `~` operator (this hould be safe with <= v0.22.x since the id is always required to be 15 characters but with v0.23+ users can adjust the length):
```js
@request.auth.id != "" &&
// leaders if multiple relation is stored as serialized json array of ids
leaders ~ @request.auth.id &&
leaders ~ user
```
